data-pipeline/src/trace_exporter.rs

```diff
@@ -191,10 +276,27 @@ impl TraceExporter {
             })
     }
 
+    /// Add all spans from the given iterator into the stats concentrator
+    fn add_spans_to_stats<'a>(&self, spans: impl Iterator<Item = &'a pb::Span>) {
+        // TODO: How do we want to react if we have an error i.e. another thread panicked with
```

Comment:
hmm good question, under what situation can a thread panic with the lock? ie. if it somehow fails to serialize a payload? or an OS level failure to lock?

Reply:
Serialization errors should not cause panic imo. It would probably be something critical and since the data can be left in an undefined state it seems reasonable to me to propagate the panic to the other threads of the `trace_exporter`